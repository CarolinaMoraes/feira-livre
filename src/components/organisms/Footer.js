import React from "react"
import styled from "styled-components";

import { FaPhone, FaBuilding, FaMousePointer, FaHandshake } from "react-icons/fa";

import Container from "components/atoms/Container";
import Grid from "components/atoms/Grid";
import Heading from "components/atoms/Heading";

const Root = styled.footer`
  background-color: #2a2a2a;
  color: #aaa;
  padding: 40px 0;

  h6 {
    color: #fff;
    margin-top: 0;
    text-transform: uppercase;
  }
`;

const IconContainer = styled.span`
  margin-right: 8px;
`;

const FooterLink = styled.a`
  color: ${props => props.theme.colors.primary.main};
  text-decoration: none;

  &:hover {
    color: ${props => props.theme.colors.primary.dark};
  }
`;

const Footer = () => (
  <Root>
    <Container>
      <Grid md={3}>
        <div>
          <Heading>
            <h6>Feira Livre</h6>
          </Heading>
          <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Fugiat vitae officia minus blanditiis quia magnam voluptatem, cumque laboriosam neque, architecto delectus eos provident illum adipisci. Consequuntur, dignissimos. Dicta, unde corrupti.</p>
        </div>
        <div>
          <Heading>
            <h6>Contatos</h6>
          </Heading>
          <p>
            <IconContainer>
              <FaPhone />
            </IconContainer>
            (11) 12345-6789</p>
          <p>
            <IconContainer>
              <FaBuilding />
            </IconContainer>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Fugiat vitae officia minus blanditiis quia magnam voluptatem</p>
        </div>
        <div>
          <Heading>
            <h6>Outros links do projeto</h6>
            <p>
              <FooterLink target="_blank" href="https://www.cozinhasolidaria.com/">
                <IconContainer>
                  <FaMousePointer />
                </IconContainer> Site do Cozinha Solid√°ria
              </FooterLink>
            </p>

            <p>
              <FooterLink target="_blank" href="https://apoia.se/cozinhasolidaria">
                <IconContainer>
                  <FaHandshake />
                </IconContainer> Apoia.se
              </FooterLink>
            </p>
          </Heading>
        </div>
      </Grid>
    </Container>
  </Root>
);


export default Footer;
