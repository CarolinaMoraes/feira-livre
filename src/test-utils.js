import React from 'react'
import ThemeProvider from "styles/ThemeProvider";
import { MemoryRouter } from 'react-router-dom';
import { render } from "@testing-library/react";

const TestProvider = ({ children }) => (
  <MemoryRouter>
    <ThemeProvider>
      {children}
    </ThemeProvider>
  </MemoryRouter>
)


export * from "@testing-library/react";

const customRender = (ui, options) => render(ui, { wrapper: TestProvider, ...options });

export { customRender as render };
