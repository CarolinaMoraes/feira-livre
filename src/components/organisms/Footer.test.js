import React from "react";
import { render } from "test-utils";

import Footer from "./Footer";

test('it renders correctly', () => {
  const { asFragment } = render(<Footer />)

  expect(asFragment()).toMatchSnapshot();
});
