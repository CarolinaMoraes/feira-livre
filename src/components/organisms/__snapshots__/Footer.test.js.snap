// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<DocumentFragment>
  .c1 {
  width: 100%;
  padding: 0 8px;
}

.c2 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c3 h1,
.c3 h2,
.c3 h3,
.c3 h4,
.c3 h5,
.c3 h6 {
  position: relative;
  margin-bottom: 25px;
  padding-bottom: 8px;
  border-bottom: 1px solid rgba(255,255,255,0.2);
}

.c3 h1::after,
.c3 h2::after,
.c3 h3::after,
.c3 h4::after,
.c3 h5::after,
.c3 h6::after {
  content: "";
  position: absolute;
  left: 0;
  bottom: 0;
  background-color: #5bcc23;
  height: 5px;
  width: 70px;
}

.c3 h1 {
  padding-bottom: 25px;
}

.c0 {
  background-color: #2a2a2a;
  color: #aaa;
  padding: 40px 0;
}

.c0 h6 {
  color: #fff;
  margin-top: 0;
  text-transform: uppercase;
}

.c4 {
  margin-right: 8px;
}

.c5 {
  color: #5bcc23;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover {
  color: #2ca352;
}

@media (min-width:576px) {
  .c1 {
    padding: 0 16px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 1140px;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c2 {
    grid-template-columns: repeat(3,1fr);
  }
}

<footer
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div>
          <div
            class="c3"
          >
            <h6>
              Feira Livre
            </h6>
          </div>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Fugiat vitae officia minus blanditiis quia magnam voluptatem, cumque laboriosam neque, architecto delectus eos provident illum adipisci. Consequuntur, dignissimos. Dicta, unde corrupti.
          </p>
        </div>
        <div>
          <div
            class="c3"
          >
            <h6>
              Contatos
            </h6>
          </div>
          <p>
            <span
              class="c4"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                />
              </svg>
            </span>
            (11) 12345-6789
          </p>
          <p>
            <span
              class="c4"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M436 480h-20V24c0-13.255-10.745-24-24-24H56C42.745 0 32 10.745 32 24v456H12c-6.627 0-12 5.373-12 12v20h448v-20c0-6.627-5.373-12-12-12zM128 76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76zm0 96c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40zm52 148h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12zm76 160h-64v-84c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v84zm64-172c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40z"
                />
              </svg>
            </span>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Fugiat vitae officia minus blanditiis quia magnam voluptatem
          </p>
        </div>
        <div>
          <div
            class="c3"
          >
            <h6>
              Outros links do projeto
            </h6>
            <p>
              <a
                class="c5"
                href="https://www.cozinhasolidaria.com/"
                target="_blank"
              >
                <span
                  class="c4"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 320 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M302.189 329.126H196.105l55.831 135.993c3.889 9.428-.555 19.999-9.444 23.999l-49.165 21.427c-9.165 4-19.443-.571-23.332-9.714l-53.053-129.136-86.664 89.138C18.729 472.71 0 463.554 0 447.977V18.299C0 1.899 19.921-6.096 30.277 5.443l284.412 292.542c11.472 11.179 3.007 31.141-12.5 31.141z"
                    />
                  </svg>
                </span>
                 Site do Cozinha Solid√°ria
              </a>
            </p>
            <p>
              <a
                class="c5"
                href="https://apoia.se/cozinhasolidaria"
                target="_blank"
              >
                <span
                  class="c4"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 640 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M434.7 64h-85.9c-8 0-15.7 3-21.6 8.4l-98.3 90c-.1.1-.2.3-.3.4-16.6 15.6-16.3 40.5-2.1 56 12.7 13.9 39.4 17.6 56.1 2.7.1-.1.3-.1.4-.2l79.9-73.2c6.5-5.9 16.7-5.5 22.6 1 6 6.5 5.5 16.6-1 22.6l-26.1 23.9L504 313.8c2.9 2.4 5.5 5 7.9 7.7V128l-54.6-54.6c-5.9-6-14.1-9.4-22.6-9.4zM544 128.2v223.9c0 17.7 14.3 32 32 32h64V128.2h-96zm48 223.9c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zM0 384h64c17.7 0 32-14.3 32-32V128.2H0V384zm48-63.9c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16c0-8.9 7.2-16 16-16zm435.9 18.6L334.6 217.5l-30 27.5c-29.7 27.1-75.2 24.5-101.7-4.4-26.9-29.4-24.8-74.9 4.4-101.7L289.1 64h-83.8c-8.5 0-16.6 3.4-22.6 9.4L128 128v223.9h18.3l90.5 81.9c27.4 22.3 67.7 18.1 90-9.3l.2-.2 17.9 15.5c15.9 13 39.4 10.5 52.3-5.4l31.4-38.6 5.4 4.4c13.7 11.1 33.9 9.1 45-4.7l9.5-11.7c11.2-13.8 9.1-33.9-4.6-45.1z"
                    />
                  </svg>
                </span>
                 Apoia.se
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
